From eb7d0883d5befaf99febb373c307c2ba5aa9a7aa Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 15 Oct 2020 16:47:24 -0700
Subject: [PATCH 0989/1921] soc: octeontx2-sdp: Fix ring info structure that is
 exchanged with host

The host side drivers have been updated of the ring info structure but
EP side wasn't. This was causing issues in negotiating ring and VF
information. This patch fixes the issue by sync'ing the structure.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: I2ec7df581483a4686678f4448ef5c10fb4f4e3ad
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/38025
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42659
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx2-sdp/sdp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2-sdp/sdp.c b/drivers/soc/marvell/octeontx2-sdp/sdp.c
index 302b9709211b..b2650d1e371c 100644
--- a/drivers/soc/marvell/octeontx2-sdp/sdp.c
+++ b/drivers/soc/marvell/octeontx2-sdp/sdp.c
@@ -39,10 +39,10 @@ union ring {
 	u64 u;
 	struct {
 		u64 dir:2;
-		u64 rpvf:4;
-		u64 rppf:6;
+		u64 rpvf:8;
+		u64 rppf:8;
 		u64 numvf:8;
-		u64 rsvd:16;
+		u64 rsvd:10;
 		u64 raz:28;
 	} s;
 };
-- 
2.31.1

